components/places/src/types.rs

```diff
@@ -133,6 +135,39 @@ impl VisitTransition {
     }
 }
 
+struct VisitTransitionSerdeVisitor;
```

Comment:
Holy cats! 🐱 TIL this is how you implement a custom deserializer, though.

Reply:
Yup. It's... not fun.